packages/features/bookings/Booker/components/AvailableTimeSlots.tsx

```diff
@@ -21,20 +23,84 @@ type AvailableTimeSlotsProps = {
  * will also fetch the next `extraDays` days and show multiple days
  * in columns next to each other.
  */
-export const AvailableTimeSlots = ({ extraDays, limitHeight, seatsPerTimeSlot }: AvailableTimeSlotsProps) => {
-  const selectedDate = useBookerStore((state) => state.selectedDate);
-  const setSelectedTimeslot = useBookerStore((state) => state.setSelectedTimeslot);
-  const setSeatedEventData = useBookerStore((state) => state.setSeatedEventData);
+export const AvailableTimeSlots = ({
+  extraDays,
+  limitHeight,
+  seatsPerTimeslot,
+  eventSlug,
+}: AvailableTimeSlotsProps) => {
   const event = useEvent();
-  const date = selectedDate || dayjs().format("YYYY-MM-DD");
   const containerRef = useRef<HTMLDivElement | null>(null);
+  const [
+    selectedDate,
+    selectedDatesAndTimes,
+    setSelectedDatesAndTimes,
+    setSelectedTimeslot,
+    setSeatedEventData,
+  ] = useBookerStore(
+    (state) => [
+      state.selectedDate,
+      state.selectedDatesAndTimes,
+      state.setSelectedDatesAndTimes,
+      state.setSelectedTimeslot,
+      state.setSeatedEventData,
+    ],
+    shallow
+  );
```

Comment:
Have just coupled the previously queried states and the new selectedDatesAndTimes states together